src/connection.rs

```diff
@@ -526,6 +546,47 @@ where
             write_key_schedule,
             Some(read_key_schedule),
         )
+        .map(|slice| (next_state, slice))
+}
+
+fn client_cert_verify<'r, Provider>(
+    key_schedule: &mut KeySchedule<Provider::CipherSuite>,
+    config: &TlsConfig,
+    crypto_provider: &mut Provider,
+    buffer: &'r mut WriteBuffer,
+) -> Result<(State, &'r [u8]), TlsError>
+where
+    Provider: CryptoProvider,
+    SignatureSize<Provider::SignatureCurve>:
+        core::ops::Add<ecdsa::der::MaxOverhead> + ArrayLength<u8>,
+    ecdsa::der::MaxSize<Provider::SignatureCurve>: ArrayLength<u8>,
+    <Provider::SignatureCurve as CurveArithmetic>::Scalar: SignPrimitive<Provider::SignatureCurve>,
+{
+    let ctx_str = b"TLS 1.3, client CertificateVerify\x00";
+    let mut msg: heapless::Vec<u8, 130> = heapless::Vec::new();
+    msg.resize(64, 0x20).map_err(|_| TlsError::EncodeError)?;
+    msg.extend_from_slice(ctx_str)
+        .map_err(|_| TlsError::EncodeError)?;
+    msg.extend_from_slice(&key_schedule.transcript_hash().clone().finalize())
+        .map_err(|_| TlsError::EncodeError)?;
+
+    // FIXME: Remove unwraps!
+    let mut signing_key = crypto_provider.signer(config.priv_key.unwrap()).unwrap();
```

Comment:
What to do here in case the signer operation fails? It could be from e.g. invalid private key.

Am I supposed to send an alert?

Reply:
I think since the server has done nothing wrong, send a close_notify alert and close the connection.